src/Datagrid/Pager.php

```diff
@@ -38,9 +38,9 @@ public function computeNbResult()
         return $countQuery->count()->getQuery()->execute();
     }
 
-    public function getResults()
+    public function getResults(): array
     {
-        return $this->getQuery()->execute();
+        return $this->getQuery()->execute()->toArray();
```

Comment:
Maybe it would be interesting to revisit `PagerInterface::getResults` to return `iterable` instead of `array`.

Reply:
Do you want to do a PR in SonataAdmin ?